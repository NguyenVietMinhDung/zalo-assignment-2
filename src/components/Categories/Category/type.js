import type { Category } from '../type';

export type Props = {...Category};
