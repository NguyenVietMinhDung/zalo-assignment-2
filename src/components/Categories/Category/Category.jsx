import React from 'react';
import { Container, Row, Col } from 'react-grid-system';
import './Category.scss';
import Product from './Product';
import { Props } from './type';

const Category = (props: Props) => {
  const { items, id, name } = props;
  return (
    <Container className="category" id={id}>
      <Row>
        <Col><h3>{name}</h3></Col>
      </Row>
      {items.map(item => <Product key={item.id} {...item} />)}
    </Container>
  );
};

export default Category;
