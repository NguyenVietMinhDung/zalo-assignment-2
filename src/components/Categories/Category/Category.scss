.category {
	display: flex;

	flex-direction: column;

	margin-bottom: 8px;

	background-color: white;
}
