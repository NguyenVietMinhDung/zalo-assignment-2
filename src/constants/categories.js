export default [
  {
    id: 'other',
    name: 'Khác',
    items: [
      {
        id: 0,
        name: {
          vi: 'Phin Inox',
          en: 'Phin Inox',
        },
        price: 80000,
        img: '270_crop_PHININOX.png',
      },
      {
        id: 1,
        name: {
          vi: 'Ly Sứ Mosaic',
          en: 'Mosaic Mug',
        },
        price: 29000,
        img: '270_crop_LYSU.png',
      },
    ],
  },
  {
    id: 'bread',
    name: 'Bánh Mì',
    items: [
      {
        id: 0,
        name: {
          vi: 'Xíu Mại',
          en: 'Pork Meatball',
        },
        price: 19000,
        img: '270_crop_BMXIUMAI.png',
      },
      {
        id: 1,
        name: {
          vi: 'Chả Lụa Xá Xíu',
          en: 'Roast Pork & Pork Sausage',
        },
        price: 19000,
        img: '270_crop_BMCHALUAXAXIU.png',
      },
      {
        id: 2,
        name: {
          vi: 'Gà Xé Nước Tương',
          en: 'Roast Chicken',
        },
        price: 19000,
        img: '270_crop_BMGAXE.png',
      },
      {
        id: 3,
        name: {
          vi: 'Thịt Nướng',
          en: 'Grilled Pork',
        },
        price: 19000,
        img: '270_crop_BMTHITNUONG.png',
      },
    ],
  },
  {
    id: 'cake',
    name: 'Bánh Ngọt',
    items: [
      {
        id: 0,
        name: {
          vi: 'Bánh Mousse Cacao',
          en: 'Cocoa Mousse Cake',
        },
        price: 29000,
        img: '270_crop_MOUSSECACAO.png',
      },
      {
        id: 1,
        name: {
          vi: 'Bánh Mousse Đào',
          en: 'Peach Mousse Cake',
        },
        price: 29000,
        img: '270_crop_MOUSSEDAO.png',
      },
      {
        id: 2,
        name: {
          vi: 'Bánh Sô-cô-la Highlands',
          en: 'Highlands Chocolate Bomb',
        },
        price: 29000,
        img: '270_crop_SOCOLAHL.png',
      },
    ],
  },
  {
    id: 'coffee',
    name: 'Cà Phê Đóng Gói',
    items: [
      {
        id: 0,
        name: {
          vi: 'Latte',
          en: 'Latte',
        },
        price: 60000,
        img: '270_crop_LATTE.png',
      },
      {
        id: 1,
        name: {
          vi: 'Cappuccino',
          en: 'Cappuccino',
        },
        price: 60000,
        img: '270_crop_CAPPUCINO.png',
      },
      {
        id: 2,
        name: {
          vi: 'Americano',
          en: 'Americano',
        },
        price: 60000,
        img: '270_crop_AMERICANO.png',
      },
      {
        id: 3,
        name: {
          vi: 'Espresso',
          en: 'Espresso',
        },
        price: 60000,
        img: '270_crop_ESPRESSO.png',
      },
      {
        id: 4,
        name: {
          vi: 'Caramel Macchiato',
          en: 'Caramel Macchiato',
        },
        price: 60000,
        img: '270_crop_CARAMEL-MACCHIATO.png',
      },
    ],
  },
  {
    id: 'Tea',
    name: 'Trà',
    items: [
      {
        id: 0,
        name: {
          vi: 'Trà Thạch Vãi',
          en: 'Lychee Tea',
        },
        price: 39000,
        img: '270_crop_TRATHACHVAI_1.png',
      },
      {
        id: 1,
        name: {
          vi: 'Trà Thạch Đào',
          en: 'Peach Jelly Tea',
        },
        price: 39000,
        img: '270_crop_TRATHACHDAO.png',
      },
      {
        id: 2,
        name: {
          vi: 'Trà Thanh Đào',
          en: 'Peach Lemongrass Tea',
        },
        price: 39000,
        img: '270_crop_TRATHANHDAO.png',
      },
      {
        id: 3,
        name: {
          vi: 'Trà Sen Vàng',
          en: 'Golden Lotus Tea',
        },
        price: 39000,
        img: '270_crop_TRASENVANG.png',
      },
    ],
  },
];
