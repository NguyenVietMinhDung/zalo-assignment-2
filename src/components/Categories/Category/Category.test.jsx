import React from 'react';
import { shallow } from 'enzyme';
import Category from './Category';
import categories from '../../../constants/categories';

const defaultProps = {
  ...categories[0],
};

describe('<Category />', () => {
  test('renders correctly', () => {
    const component = shallow(<Category {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
});
