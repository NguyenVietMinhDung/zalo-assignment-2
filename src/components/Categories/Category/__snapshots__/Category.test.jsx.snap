// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Category /> renders correctly 1`] = `
<Container
  className="category"
  component="div"
  fluid={false}
  id="other"
  lg={false}
  md={false}
  sm={false}
  style={Object {}}
  xl={false}
  xs={false}
  xxl={false}
>
  <Row
    align="normal"
    component="div"
    debug={false}
    gutterWidth={null}
    justify="start"
    nogutter={false}
    nowrap={false}
    style={Object {}}
  >
    <Col
      component="div"
      debug={false}
      lg={null}
      md={null}
      offset={Object {}}
      pull={Object {}}
      push={Object {}}
      sm={null}
      style={Object {}}
      width={null}
      xl={null}
      xs={null}
      xxl={null}
    >
      <h3>
        Khác
      </h3>
    </Col>
  </Row>
  <Product
    description="Phin Inox"
    id={0}
    img="270_crop_PHININOX.png"
    key="0"
    name="Phin Inox"
    price={80000}
  />
  <Product
    description="Mosaic Mug"
    id={1}
    img="270_crop_LYSU.png"
    key="1"
    name="Ly Sứ Mosaic"
    price={29000}
  />
</Container>
`;
